src/common/Error/index.js

```diff
@@ -1,14 +1,21 @@
 import React from 'react';
-import { useHistory } from 'react-router-dom';
+import { useHistory, useLocation } from 'react-router-dom';
 import { Container, StyledErrorIcon, Header, Instruction, StyledButton } from './styled';
 import { setError } from '../../features/movies/MovieDetails/movieSlice';
 
 const Error = () => {
     const history = useHistory();
+    const location = useLocation();
 
     const handleBackToHomepage = () => {
         setError(null); // Reset the error state before redirecting
```

Comment:
this line only resets error state in movieDetails Slice. when we get an error on any different page we will be stuck with it. We can do this this way if we don't import the setError action but pass it as a prop to `<Error>` component form each Page - then it will update the state slice of that Page

Reply:
Thank you for bringing this to my attention ✨ I agree with you that the current implementation of resetting the error state only in the `MovieDetails` slice could lead to issues when an error occurs on a different page.

I understand the suggestion to pass the `setError` action as a prop to the `<Error>` component on each page, allowing the update of the state slice for that specific page.

I am happy to make these changes. Thanks again for the valuable feedback! 😊